### PR TITLE
Lpf reader

### PR DESCRIPTION
**Pull request by pluAtAnsys**

### CODE REVIEW

File: ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    ----------
+    data_result : list
+        a list of data in format: [x, y, optical power]
+
+    Returns
+    -------
+
+
+    """
+    x = []
```
Comment (StefanThoene) [inline ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py:150]: why not create an xmp map??

File: ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+
+    """
+    data = []
+    for selection in range(5):
+        file_name = getfilename("*.lpf")
+        if file_name == "":
+            print("Complete, no file has been selected")
+        else:
+            my_lpf = DpfLpfReader(231)
```
Comment (StefanThoene) [inline ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py:181]: don't hard code version number

File: ansys_optical_automation/post_process/dpf_lpf_reader.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import os
+import sys
+
+
+class DpfLpfReader:
+    def __init__(self, speos_version):
+        if speos_version not in [231]:
```
Comment (StefanThoene) [inline ansys_optical_automation/post_process/dpf_lpf_reader.py:7]: adjust version list 232 is released

**Review by StefanThoene — CHANGES_REQUESTED** [review]
hi there are some minor refactors to do and it would make sense to quickly review the refractive power computation

File: ansys_optical_automation/scdm_core/utils.py [hunk 4]
```diff
@@ -120,6 +120,38 @@ def vector_dot_product(vector1, vector2):
...
+
+    Returns
+    -------
+    float
+
+    """
+    return rad * 180.0 / math.pi
+
+
+def radiance(deg):
```
Comment (StefanThoene) [inline ansys_optical_automation/scdm_core/utils.py:144]: typo

File: ansys_optical_automation/scdm_core/utils.py [hunk 5]
```diff
@@ -120,6 +120,38 @@ def vector_dot_product(vector1, vector2):
         [x,y,z]
     Returns
     -------
-    int
+    float
     """
     return vector1[0] * vector2[0] + vector1[1] * vector2[1] + vector1[2] * vector2[2]
+
+
+def degree(rad):
```
Comment (StefanThoene) [inline ansys_optical_automation/scdm_core/utils.py:128]: why not use inbuild numpy conversion

File: ansys_optical_automation/scdm_core/utils.py (hunk 4 above)
Comment (StefanThoene) [reply ansys_optical_automation/scdm_core/utils.py:144]: why not use inbuild numpy conversion

File: ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py [hunk 6]
```diff
@@ -156,16 +180,33 @@ def plot_result(data_result):
...
     for item in data_result:
-        x.append(item.start_position[0])
-        y.append(item.start_position[1])
-        if item.optical_power > 0.1:
-            value.append(0)
-        else:
-            value.append(item.optical_power)
-    plt.scatter(x, y, c=value, s=20)
-    plt.colorbar()
+        x.append(item.start_position[1])
```
Comment (StefanThoene) [inline ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py:188]: why no build s Speos xmp?

**Review by StefanThoene — CHANGES_REQUESTED** [review]

File: ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py (hunk 1 above)
Comment (pluAtAnsys) [reply ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py:181]: added a function to write result into XMP file

File: ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py (hunk 6 above)
Comment (pluAtAnsys) [reply ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py:188]: added a function to write result into XMP file

File: ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py (hunk 2 above)
Comment (pluAtAnsys) [reply ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py]: updated for users to select speos version

File: ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+            )
+            lpf_out = [
+                trace.LastDirection.Get(0),
+                trace.LastDirection.Get(1),
+                math.sqrt(1 - trace.LastDirection.Get(0) ** 2 - trace.LastDirection.Get(1) ** 2),
+            ]
+            alpha = math.acos(vector_dot_product(lpf_in, lpf_out))
+            if lpf_out[1] < 0:
+                alpha = -alpha
+            else:
```
[the remark this answers is not in the document]
Comment (pluAtAnsys) [reply ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py]: removed. It was a wrong calculation method

File: ansys_optical_automation/scdm_core/utils.py (hunk 4 above)
Comment (pluAtAnsys) [reply ansys_optical_automation/scdm_core/utils.py:146]: updated to numpy library method

File: ansys_optical_automation/scdm_core/utils.py (hunk 5 above)
Comment (pluAtAnsys) [reply ansys_optical_automation/scdm_core/utils.py:130]: updated to numpy library method

**Review by StefanThoene — APPROVED** [review]

**Pull request merged**